src/nvim/message.c

```diff
@@ -249,35 +249,37 @@ bool msg(const char *s, const int hl_id)
   return msg_hl_keep(s, hl_id, false, false);
 }
 
-/// Similar to msg_outtrans, but support newlines and tabs.
-void msg_multiline(const char *s, int hl_id, bool check_int, bool hist, bool *need_clear)
+/// Similar to msg_outtrans_len, but support newlines and tabs.
+void msg_multiline(const char *str, int hl_id, bool check_int, bool hist, bool *need_clear, int len)
```

Comment:
Could change it to `String str`. Callers with unknown length can use `cstr_as_string`  (works without copy)